File: client/src/app/shared/models/Change.ts
import AvailableAction from './enums/AvailableAction';
import TodoTask from './TodoTask';
import User from './User';

export default interface Change {
  id: string;
  creator: User;
  task: TodoTask | undefined;
  actionType: AvailableAction;
  creationDate: Date;
  taskTitle: string | undefined;
  projectId: string;
  isRead: boolean;
}

File: Core/Entities/Change.cs
namespace Core.Entities;
public class Change : BaseEntity
{
    public int CreatorId { get; set; }
    public User Creator { get; set; } = null!;
    public Guid? TaskId { get; set; }
    public TodoTask? Task { get; set; }
    public string ActionType { get; set; } = null!;
    public string TaskTitle { get; set; } = null!;
    public Guid ProjectId { get; set; }
    public Project Project { get; set; } = null!;
    public ICollection<UserChange>? UserChanges { get; set; }
}
